internal/cmd/ssh.go

```diff
@@ -253,8 +253,12 @@ func provisionSSHKey(ctx context.Context, opts provisionSSHKeyOptions) (string,
 	}
 
 	keysDir := filepath.Join(opts.infraSSHDir, "keys")
-	existingKeys := matchingPublicKeys(keysCfg, opts.hostConfig, org.ID)
-	for i, existing := range existingKeys {
+	for i := 0; i < len(keysCfg.Keys); {
+		existing := keysCfg.Keys[i]
+		if !publicKeyMatches(existing, opts.hostConfig, org.ID) {
+			i++
```

Comment:
```suggestion
                        // this public key is not managed by infra, we can proceed to the next index
			i++
```
Does this comment describe what is happening here? I was a bit confused.

Reply:
Almost, it's very likely still managed by infra, but it's for a different infra server, org, or user. I'll add the comment